core/vm/contracts.go

```diff
@@ -93,7 +93,8 @@ func ecrecoverFunc(in []byte) []byte {
 	vbig := common.Bytes2Big(in[32:64])
 	v := byte(vbig.Uint64())
 
-	if !crypto.ValidateSignatureValues(v, r, s) {
+	// tighter sig s values in homestead only apply to tx sigs
```

Comment:
@obscuren It's not wrong. https://github.com/ethereum/EIPs/blob/master/EIPS/eip-2.mediawiki does not specify this for EC Recover precompiled contract. It only specifies it for txs.

If we want to apply the same rule to the precompiled contract then the EIP needs to be updated.